src/main/groovy/org/beryx/runtime/data/TargetPlatform.groovy

```diff
@@ -16,60 +16,42 @@
 package org.beryx.runtime.data
 
 
-import groovy.transform.CompileStatic
 import org.beryx.runtime.util.JdkUtil
+import org.gradle.api.Named
 import org.gradle.api.Project
+import org.gradle.api.internal.provider.DefaultProvider
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 
-@CompileStatic
-class TargetPlatform implements Serializable {
+abstract class TargetPlatform implements Serializable, Named {
     private static final Logger LOGGER = Logging.getLogger(TargetPlatform.class)
+    private Project project
 
-    transient private final Project project
-    final String name
-    private Serializable jdkHome
-    List<String> options = []
+    abstract Property<String> getJdkHome()
+    abstract ListProperty<String> getOptions()
 
-    TargetPlatform(Project project, String name, String jdkHome = '', List<String> options = []) {
+    void setProject(Project project) {
         this.project = project
-        this.name = name
-        this.@jdkHome = jdkHome
-        this.options.addAll(options)
-    }
-    String getJdkHome() {
-        (this.@jdkHome == null) ? null : this.@jdkHome.toString()
     }
 
-    void setJdkHome(Serializable jdkHome) {
-        this.@jdkHome = jdkHome
+    void setJdkHome(Provider<String> jdkHome) {
+        this.jdkHome.set(jdkHome)
     }
 
     void addOptions(String... opts) {
-        opts.each { String opt -> options.add(opt) }
-    }
-
-    private static class LazyString implements Serializable {
-        final Closure<String> closure
-        LazyString(Closure<String> closure) {
-            this.closure = closure
-        }
-
-        @Lazy String string = closure.call()
-
-        @Override
-        String toString() {
-            string
-        }
+        this.options.addAll(opts)
     }
 
-    LazyString jdkDownload(String downloadUrl, Closure downloadConfig = null) {
+    Provider<String> jdkDownload(String downloadUrl, Closure downloadConfig = null) {
```

Comment:
This is another changed I made, since this was an internal type. It changed the approach to rely on `Provider<String>` but in practice it should be equivalent